synapseclient/table.py

```diff
@@ -777,6 +821,51 @@ def __init__(self, values, rowId=None, versionNumber=None, etag=None):
             self.etag = etag
 
 
+class PartialRow(DictObject):
+    """
+    A partial update for a single row.
+    Note "values" is a :py:class:`dict` where:
+        - key is the columnId to the desired row
+        - value is the new desired value for that column
+    The easiest way to know which columnIds to use is to check the `headers` of your most recent query:"
+        query_result = syn.TableQuery("SELECT * FROM syn123")
+        name_to_col_id = {col.name:col.id for col in query_results.headers}
+
+        row_id = #you to set this yourself#
+
+        values_to_change = {'My Row Name': "new value",
+                            'My Other Row': 42}
+        partial_row = PartialRow(values_to_change, row_id, nameToColumnId=name_to_col_id)
+
+    :param values: A dict where:
+                - key is the columnId to the desired row
+                - value is the new desired value for that column
+    :param rowId: THe id of the row to be updated
+    :param etag: used for updating File/Project Views. Not necessary if updating data Tables
```

Comment:
Rephrase, a data Table is not clear - 'only needed for File and Project Views'  Also add link to these classes.